Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/risklevel/ui/TestRiskLevelCalculationFragmentCWAViewModel.kt

```diff
@@ -0,0 +1,291 @@
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+class TestRiskLevelCalculationFragmentCWAViewModel @AssistedInject constructor(
+    @Assisted private val handle: SavedStateHandle,
+    @Assisted private val exampleArg: String?,
+    private val context: Context, // App context
+    private val exposureNotificationClient: ExposureNotificationClient
+) : CWAViewModel() {
+
+    val startLocalQRCodeScanEvent = SingleLiveEvent<Unit>()
+    val riskLevelResetEvent = SingleLiveEvent<Unit>()
+    val apiKeysProvidedEvent = SingleLiveEvent<DiagnosisKeyProvidedEvent>()
+    val riskScoreState = MutableLiveData<RiskScoreState>(RiskScoreState())
+
+    init {
+        Timber.d("CWAViewModel: %s", this)
+        Timber.d("SavedStateHandle: %s", handle)
+        Timber.d("Example arg: %s", exampleArg)
+    }
+
+    fun retrieveDiagnosisKeys() {
+        viewModelScope.launch {
+            try {
+                RetrieveDiagnosisKeysTransaction.start()
+                calculateRiskLevel()
+            } catch (e: TransactionException) {
+                e.report(ExceptionCategory.INTERNAL)
+            }
+        }
+    }
+
+    fun calculateRiskLevel() {
+        viewModelScope.launch {
+            try {
+                RiskLevelTransaction.start()
+            } catch (e: TransactionException) {
+                e.report(ExceptionCategory.INTERNAL)
+            }
+        }
+    }
+
+    fun resetRiskLevel() {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                try {
+                    // Preference reset
+                    SecurityHelper.resetSharedPrefs()
+                    // Database Reset
+                    AppDatabase.reset(context)
+                    // Export File Reset
+                    FileStorageHelper.getAllFilesInKeyExportDirectory().forEach { it.delete() }
+
+                    LocalData.lastCalculatedRiskLevel(RiskLevel.UNDETERMINED.raw)
+                    LocalData.lastSuccessfullyCalculatedRiskLevel(RiskLevel.UNDETERMINED.raw)
+                    LocalData.lastTimeDiagnosisKeysFromServerFetch(null)
+                    LocalData.googleApiToken(null)
+                } catch (e: Exception) {
+                    e.report(ExceptionCategory.INTERNAL)
+                }
+            }
+            RiskLevelTransaction.start()
+            riskLevelResetEvent.postValue(Unit)
+        }
+    }
+
+    data class RiskScoreState(
+        val riskScoreMsg: String = "",
+        val backendParameters: String = "",
+        val exposureSummary: String = "",
+        val formula: String = "",
+        val fullConfig: String = "",
+        val exposureInfo: String = ""
+    )
+
+    fun startENFObserver() {
+        viewModelScope.launch {
+            try {
+                var workState = riskScoreState.value!!
```

Comment:
shouldn't we handle a null value more explicit?

Reply:
It can never be null as we initialize with a value and the data type is not nullable:
```kotlin
val riskScoreState = MutableLiveData<RiskScoreState>(RiskScoreState())
```